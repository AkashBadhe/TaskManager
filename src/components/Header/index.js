import React from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';
import {FaGithub} from 'react-icons/fa';

const Wrapper = styled.div`
  height: 50px;
  background-color: black;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 0 1.1rem;
  color: white;
`;

const HomeLink = styled(Link)`
  text-decoration: none;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const HomeLinkLogo = styled.div`
  font-size: 2rem;
  padding: 4px;
  color: white;
`;

const GithubLink = styled.a`
  color: white !important;
  height: 50px;
  width: 17px;
  display: flex;
  flex-direction: column;
  justify-content: center;

  &:hover,
  &:focus,
  &:active {
    opacity: 0.8;
  }
`;

const Header = () => {
  return (
    <Wrapper>
      <HomeLink to="/">
        <HomeLinkLogo>
          <span role="img" aria-label="task">
            Task Manager
          </span>
        </HomeLinkLogo>
      </HomeLink>
      <GithubLink href="https://github.com/AkashBadhe/Task-Management-App.git" target="_blank">
        <FaGithub size={20} />
      </GithubLink>
    </Wrapper>
  );
};

export default Header;
