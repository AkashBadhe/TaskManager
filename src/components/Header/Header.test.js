import React from "react";
import { shallow } from "enzyme";
import Header from "./index";

describe("Header", () => {
  let header = shallow(<Header />);
  it("Should mount `Header`", () => {
    expect(header).toMatchSnapshot();
  });

  it("Should have `#homeLink`", () => {
    expect(header.find("#homeLink").exists()).toBe(true);
  });

  it("Should have `#logo`", () => {
    expect(header.find("#logo").exists()).toBe(true);
  });

  it("Should have `#githubLink`", () => {
    expect(header.find("#githubLink").exists()).toBe(true);
  });
});
