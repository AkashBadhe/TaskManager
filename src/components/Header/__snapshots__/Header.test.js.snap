// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should mount \`Header\` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Unknown
          id="homeLink"
          to="/"
>
          <Unknown>
                    <span
                              aria-label="task"
                              id="logo"
                              role="img"
                    >
                              Task Manager
                    </span>
          </Unknown>
</Unknown>,
        <Unknown
          href="https://github.com/AkashBadhe/TaskManager"
          id="githubLink"
          target="_blank"
>
          <FaGithub
                    size={20}
          />
</Unknown>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Unknown>
            <span
                        aria-label="task"
                        id="logo"
                        role="img"
            >
                        Task Manager
            </span>
</Unknown>,
          "id": "homeLink",
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              aria-label="task"
              id="logo"
              role="img"
>
              Task Manager
</span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "task",
              "children": "Task Manager",
              "id": "logo",
              "role": "img",
            },
            "ref": null,
            "rendered": "Task Manager",
            "type": "span",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "baseHash": 1842686996,
              "componentId": "sc-AxiKw",
              "isStatic": false,
              "rules": Array [
                "
  font-size: 2rem;
  padding: 4px;
  color: white;
",
              ],
              "staticRulesId": "",
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "sc-AxiKw",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "baseHash": 640081696,
            "componentId": "sc-AxirZ",
            "isStatic": false,
            "rules": Array [
              "
  text-decoration: none;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
",
            ],
            "staticRulesId": "",
          },
          "displayName": "Styled(Link)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-AxirZ",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FaGithub
            size={20}
/>,
          "href": "https://github.com/AkashBadhe/TaskManager",
          "id": "githubLink",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "baseHash": 1845756808,
            "componentId": "sc-AxhCb",
            "isStatic": false,
            "rules": Array [
              "
  color: white !important;
  height: 50px;
  width: 17px;
  display: flex;
  flex-direction: column;
  justify-content: center;

  &:hover,
  &:focus,
  &:active {
    opacity: 0.8;
  }
",
            ],
            "staticRulesId": "",
          },
          "displayName": "styled.a",
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-AxhCb",
          "target": "a",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "baseHash": 400283751,
        "componentId": "sc-AxjAm",
        "isStatic": false,
        "rules": Array [
          "
  height: 50px;
  background-color: black;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 0 1.1rem;
  color: white;
",
        ],
        "staticRulesId": "",
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "sc-AxjAm",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown
            id="homeLink"
            to="/"
>
            <Unknown>
                        <span
                                    aria-label="task"
                                    id="logo"
                                    role="img"
                        >
                                    Task Manager
                        </span>
            </Unknown>
</Unknown>,
          <Unknown
            href="https://github.com/AkashBadhe/TaskManager"
            id="githubLink"
            target="_blank"
>
            <FaGithub
                        size={20}
            />
</Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Unknown>
              <span
                            aria-label="task"
                            id="logo"
                            role="img"
              >
                            Task Manager
              </span>
</Unknown>,
            "id": "homeLink",
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                aria-label="task"
                id="logo"
                role="img"
>
                Task Manager
</span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "task",
                "children": "Task Manager",
                "id": "logo",
                "role": "img",
              },
              "ref": null,
              "rendered": "Task Manager",
              "type": "span",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "baseHash": 1842686996,
                "componentId": "sc-AxiKw",
                "isStatic": false,
                "rules": Array [
                  "
  font-size: 2rem;
  padding: 4px;
  color: white;
",
                ],
                "staticRulesId": "",
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "shouldForwardProp": undefined,
              "styledComponentId": "sc-AxiKw",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "baseHash": 640081696,
              "componentId": "sc-AxirZ",
              "isStatic": false,
              "rules": Array [
                "
  text-decoration: none;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
",
              ],
              "staticRulesId": "",
            },
            "displayName": "Styled(Link)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "sc-AxirZ",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FaGithub
              size={20}
/>,
            "href": "https://github.com/AkashBadhe/TaskManager",
            "id": "githubLink",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 20,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "baseHash": 1845756808,
              "componentId": "sc-AxhCb",
              "isStatic": false,
              "rules": Array [
                "
  color: white !important;
  height: 50px;
  width: 17px;
  display: flex;
  flex-direction: column;
  justify-content: center;

  &:hover,
  &:focus,
  &:active {
    opacity: 0.8;
  }
",
              ],
              "staticRulesId": "",
            },
            "displayName": "styled.a",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "sc-AxhCb",
            "target": "a",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "baseHash": 400283751,
          "componentId": "sc-AxjAm",
          "isStatic": false,
          "rules": Array [
            "
  height: 50px;
  background-color: black;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 0 1.1rem;
  color: white;
",
          ],
          "staticRulesId": "",
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "shouldForwardProp": undefined,
        "styledComponentId": "sc-AxjAm",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
